src/fides/api/ops/api/v1/endpoints/dataset_endpoints.py

```diff
@@ -198,7 +259,13 @@ def patch_datasets(
             "dataset": dataset.dict(),
```

Comment:
Validation on this endpoint makes sure data categories on the dataset exist in the database.  Because we're accessing the database, it's done outside of a typical pydantic validator.  If this endpoint goes away, we need a new place to stick this. Does the existing ctl_datasets endpoint have this validation?

Reply:
CC: @ThomasLaPiana. You might be the right person to ask about the ctl side of this

Reply:
OK so it looks like the existing crud endpoints don't do this.  The tricky bit is that the crud endpoints are very generic, it's blocks of code that applies to updating an entire set of resources.  Added a note to look into the best place to put this in the next ticket https://github.com/ethyca/fides/issues/1763